rust/arrow/src/array.rs

```diff
@@ -249,8 +263,113 @@ impl<T: ArrowNumericType> PrimitiveArray<T> {
     }
 }
 
+impl<T: ArrowTemporalType + ArrowNumericType> PrimitiveArray<T>
+where
+    i64: std::convert::From<T::Native>,
+{
+    /// Returns value as a chrono `NaiveDateTime`, handling time resolution
+    ///
+    /// If a data type cannot be converted to `NaiveDateTime`, a `None` is returned.
+    /// A valid value is expected, thus the user should first check for validity.
+    /// TODO: extract constants into static variables
```

Comment:
can we extract the constants in this PR?

Reply:
Looks like I was benefiting a lot from implicit conversions between signed and unsigned, adding constants. I thought I could quickly extract the constants, but I have to spend a bit more time and reason about my casting. I'll do that later or tomorrow

```rust
error[E0277]: cannot multiply `i32` to `u32`
   --> arrow\src\array.rs:293:43
    |
293 |                 (v % MILLISECONDS) as u32 * MICROSECONDS,
    |                                           ^ no implementation for `u32 * i32`
```